notation.go

```diff
@@ -35,6 +37,15 @@ type SignOptions struct {
 
 	// Metadata is optional for the artifact to be signed.
 	Metadata Metadata
+
+	// TSA is the TimeStamp Authority to timestamp the resulted signature if present.
+	TSA timestamp.Timestamper
+
+	// TSAVerifyOptions is the verify option to verify the fetched timestamp signature.
+	// The `Intermediates` in the verify options will be ignored and re-contrusted using
+	// the certificates in the fetched timestamp signature.
+	// An empty list of `KeyUsages` in the verify options implies ExtKeyUsageTimeStamping.
+	TSAVerifyOptions x509.VerifyOptions
```

Comment:
We need to verify the timestamp signature for security pupose since the timestamp signature is usually transferred via HTTP not HTTPS. Anyone can modify the timestamp signature if we don't do verification. In that case, an adversary can replace the signature with a signature signed by an unknown authority. If the notation signer blindly attachs the bad timestamp signature to the artifact signature, the signaturte will not be verifiable after expiry and impact the expected availalbility.

Reply:
We don't want signer-provider or verification-provider to override the [TSAVerifyOptions](https://pkg.go.dev/crypto/x509#VerifyOptions) because then user can override this value and start using non time-stamping cert(like ssl) for time-stamping.

Also as per [rfc3161#section-2.3 ](https://datatracker.ietf.org/doc/html/rfc3161#section-2.3)
`The corresponding certificate MUST contain only one instance of the extended key usage field extension as defined in [RFC2459] Section 4.2.1.13 with KeyPurposeID having value:
 id-kp-timeStamping.`

I would suggest to start with not exposing `TSAVerifyOptions` and then later if need arises we can expose this option with sane defaults.